crates/typst/src/visualize/image/mod.rs

```diff
@@ -190,6 +190,22 @@ fn layout_image(
         Smart::Auto => determine_format(elem.path().as_str(), data).at(span)?,
     };
 
+    // Warn the user if the image contains a foreign object. Not perfect
+    // because the svg could also be encoded, but that's an edge case.
+    if format == ImageFormat::Vector(VectorFormat::Svg) {
+        let has_foreign_object =
+            data.as_str().map(|s| s.contains("<foreignObject")).unwrap_or(false);
```

Comment:
```suggestion
            data.as_str().is_some_and(|s| s.contains("<foreignObject"));
```
sorry, but my refactoring pattern matching brain can never stop 🙈